core/validation/initial_check.py

```diff
@@ -0,0 +1,94 @@
+"""Initial Pre-transformation Validation of sheets to be ingested, looking at
+the form version, reporting round, sheets, fund type, and place names, and ensuring
+that the input for these is valid for the specific round of data we are ingesting.
+"""
+import numpy as np
+import pandas as pd
+
+import core.validation.failures as vf
+from core.const import EXPECTED_ROUND_THREE_SHEETS
+
+
+def extract_round_three_submission_details(workbook: dict[str, pd.DataFrame]) -> dict[str, list[str]]:
+    """
+    Extract submission details for round three from the given workbook.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas DataFrames.
+    :return: A dictionary containing the extracted submission details.
+    """
+    details_dict = {}
+
+    missing_sheets = check_missing_sheets(EXPECTED_ROUND_THREE_SHEETS, workbook)
+
+    if len(missing_sheets) > 0:
+        return {"Missing Sheets": missing_sheets}
+
+    sheet_a1 = workbook.get("1 - Start Here")
+    if sheet_a1 is not None:
+        details_dict["Form Version"] = [
+            sheet_a1.iloc[6][1],
+            "Town Deals and Future High Streets Fund Reporting Template (v3.0)",
+        ]
+        details_dict["Reporting Period"] = [sheet_a1.iloc[4][1], "1 October 2022 to 31 March 2023"]
+
+    sheet_a2 = workbook.get("2 - Project Admin")
+    if sheet_a2 is not None:
+        details_dict["Fund Type"] = [sheet_a2.iloc[5][4], "Town_Deal", "Future_High_Street_Fund"]
+        details_dict["Place Name"] = [sheet_a2.iloc[6][4]]
+
+    return details_dict
+
+
+def pre_transformation_check(submission_details: dict[str, list[str]]) -> list[vf.ValidationFailure]:
+    """
+    Perform pre-transformation checks on the given workbook.
+
+    :param submission_details: A dictionary containing submission details.
+    :return: A list of validation failures encountered during pre-transformation checks.
+    """
+
+    if "Missing Sheets" in submission_details:
+        return [vf.EmptySheetFailure(empty_sheet=empty_sheet) for empty_sheet in submission_details["Missing Sheets"]]
+
+    pre_transformation_failures = [
+        check_form_input(key, value)
+        for key, value in submission_details.items()
+        if check_form_input(key, value) is not None
+    ]
```

Comment:
This code will unnecessary call `check_form_input` a second time. Better to do something like this:
```
failures = []
for value_descriptor, values in submission_details.items():
    if failure := check_form_input(value_descriptor, values):
        failures.append(failure)
```